src/main/java/io/quarkus/registry/app/maven/cache/MavenCacheFilter.java

```diff
@@ -37,9 +37,12 @@
     @ConfigProperty(name = "maven.cache.header.cache-control", defaultValue = "no-cache") // Check for etag everytime
     Instance<String> headerCacheControl;
 
+    @ConfigProperty(name = "maven.cache.enabled", defaultValue = "true")
+    Instance<Boolean> enabled;
+    
     @Override
     public void filter(ContainerRequestContext containerRequestContext) throws IOException {
-        if (isMavenGetRequest(containerRequestContext)) {
+        if (isMavenGetRequest(containerRequestContext) && enabled.get()) {
```

Comment:
What about the filter method from `ContainerResponseFilter`? Should it also check for this flag as well?

Reply:
Well, assuming we can enable/disable in runtime, we can still cache the values, but not read from it, then when we want we can enable and have a hot cache... but only of we can enable without restart.

Reply:
I am not sure we can enable it without restarting. Last time I checked `ConfigSources` were immutable during runtime.

Reply:
OK, this is a use case to have the option to not have that :) Let me also check there then . Hold on